src/main/java/net/glowstone/util/GlowTravelAgent.java

```diff
@@ -0,0 +1,319 @@
+package net.glowstone.util;
+
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import net.glowstone.EventFactory;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.TravelAgent;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.world.PortalCreateEvent;
+
+@Accessors(chain = true)
+public class GlowTravelAgent implements TravelAgent {
+
+    @Getter
+    @Setter
+    private int searchRadius = 128;
+    @Getter
+    private int creationRadius = 16;
+    @Accessors(chain = false)
+    @Setter
+    private boolean canCreatePortal = true;
+    private World world;
+
+    /**
+     * Construct a new TravelAgent.
+     * @param world the world this agent operates in
+     */
+    public GlowTravelAgent(World world) {
+        if (world.getEnvironment() == World.Environment.THE_END) {
+            throw new UnsupportedOperationException("End Portals are not supported yet.");
+        }
+        this.world = world;
+    }
+
+    @Override
+    public Location findPortal(Location destination) {
+        Location minLoc = null;
+        double minDistance = Double.MAX_VALUE;
+
+        final int blockX = destination.getBlockX();
+        final int blockZ = destination.getBlockZ();
+
+        for (int x = blockX - searchRadius; x < (blockX + searchRadius); x++) {
+            for (int z = blockZ - searchRadius; z < (blockZ + searchRadius); z++) {
+                for (int y = 127; y >= 0; y--) {
```

Comment:
Just fyi: The start y value should be depending on the world type like it is in Vanilla Minecraft (127 for the nether, 255 for the rest)

Reply:
Thanks, totally missed that one